src/main/java/edu/uw/edm/edmzuulproxy/certificateauthorizer/service/impl/CertificateAuthorizerServiceImpl.java

```diff
@@ -31,63 +33,64 @@
 
-        final boolean methodIsAuthorized = methodIsAuthorized(httpMethod, certificateAuthorization);
-        final boolean uriIsAuthorized = uriIsAuthorized(uri, certificateAuthorization);
-        final boolean uwGroupsMatch = uwGroupsMatch(user, certificateAuthorization);
+        final boolean authorized = methodIsAuthorized(httpMethod, certificateAuthorizationDAO) &&
+                uriIsAuthorized(uri, certificateAuthorizationDAO)
+                && uwGroupsMatch(user, certificateAuthorizationDAO);
 
-
-        final boolean authorized = methodIsAuthorized &&
-                uriIsAuthorized &&
-                uwGroupsMatch;
-
-        log.trace("method : {} , uri : {}, uwGroups: {} => result : {}", methodIsAuthorized, uriIsAuthorized, uwGroupsMatch, authorized);
         return authorized;
 
     }
 
-    private boolean uwGroupsMatch(User user, CertificateAuthorization certificateAuthorization) {
-        return certificateAuthorization.getUwGroups() == null || certificateAuthorization.getUwGroups().contains(UW_GROUPS_WILDCARD) || userIsMemberOfOneGroup(user, certificateAuthorization);
+    private boolean uwGroupsMatch(User user, CompiledCertificateAuthorization certificateAuthorization) {
+        return certificateAuthorization.getUwGroups() == null
+                || certificateAuthorization.getUwGroups().contains(UW_GROUPS_WILDCARD)
+                || userIsMemberOfOneGroup(user, certificateAuthorization);
     }
 
-    private boolean userIsMemberOfOneGroup(User user, CertificateAuthorization certificateAuthorization) {
-        //TODO
-        return false;
-    }
+    private boolean userIsMemberOfOneGroup(User user, CompiledCertificateAuthorization certificateAuthorization) {
+
 
-    private boolean uriIsAuthorized(String uri, CertificateAuthorization certificateAuthorization) {
-        final Pattern pattern = Pattern.compile(certificateAuthorization.getUriRegex());
+        return user!=null && user.getAuthorities()
+                .stream()
+                .map(GrantedAuthority::getAuthority)
+                .anyMatch(authority -> certificateAuthorization.getUwGroups().contains(authority));
+    }
 
-        return pattern.matcher(uri).matches();
+    private boolean uriIsAuthorized(String uri, CompiledCertificateAuthorization certificateAuthorization) {
+        return certificateAuthorization.getUriRegex().matcher(uri).matches();
     }
 
-    private boolean methodIsAuthorized(HttpMethod httpMethod, CertificateAuthorization authorization) {
-        return authorization.getHttpMethods().contains(HTTP_METHOD_WILDCARD) || authorization.getHttpMethods().contains(httpMethod.name());
+    private boolean methodIsAuthorized(HttpMethod httpMethod, CompiledCertificateAuthorization authorization) {
+        return authorization.getHttpMethods().contains(HTTP_METHOD_WILDCARD)
```

Comment:
should we check that getHttpMethods is not null?

Reply:
I'm now checking for null ( or empty ) of all the mandatory fields  in `edu.uw.edm.edmzuulproxy.certificateauthorizer.service.impl.CertificateAuthorizationRetriever#toCompiledCertificateAuthorization` )